### PR TITLE
Simplify params model state for com_contact, com_newsfeeds

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue # .

### Summary of Changes
Use $app->getParams() to get $params variable. Less code, consistent with other model classes.

### Testing Instructions
Code review should be enough.

### CODE REVIEW

File: components/com_contact/src/Model/CategoryModel.php [hunk 1]
```diff
@@ -267,20 +267,9 @@ protected function getListQuery()
 	protected function populateState($ordering = null, $direction = null)
 	{
 		$app = Factory::getApplication();
-		$params = ComponentHelper::getParams('com_contact');
```
Comment (Quy) [inline components/com_contact/src/Model/CategoryModel.php]: Please remove `use Joomla\CMS\Component\ComponentHelper;`.

Comment (joomdonation) [reply components/com_contact/src/Model/CategoryModel.php]: Thanks @Quy.

**Comment by wilsonge** [comment]
https://github.com/joomla/joomla-cms/blob/staging/components/com_content/models/category.php#L117-L127 We have the same code in com_content at the moment.

Newsfeeds doesn't even merge the menu specific items :) https://github.com/joomla/joomla-cms/blob/staging/components/com_newsfeeds/models/category.php#L197

Overall I think this is right - we just need to get everything consistent at once.

**Comment by joomdonation** [comment]
@wilsonge We handle it different in com_content, thus I made a separate PR https://github.com/joomla/joomla-cms/pull/34892 for com_content.

For newsfeeds, yes, we have it in different way, too. So 3 components, should work in the same way but we handle it different in each component :D. I will add the same change to com_newsfeeds into this PR

**Comment by joomdonation** [comment]
OK. I use similar code for com_newsfeeds. Assume it is accepted, we will have consistent code for the three components :).

**Comment by HLeithner** [comment]
This pull request has automatically rebased to 4.2-dev.

**Review by rdeutz — APPROVED** [review]

**Review by drmenzelit — APPROVED** [review]

**Pull request merged**